File: problem-sets/problem-set-2/readability/readability.c
#include "cs50.h"
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int letter_count = 0;
int words_count = 1;
int sentences_count = 0;

int main(void)
{
    // Get user input
    string text = get_string("Text: ");

    // Call functions
    count_letters(text);
    count_words(text);
    count_sentences(text);

    // Calculate L and S
    float l = ((float) letter_count / words_count) * 100;
    float s = ((float) sentences_count / words_count) * 100;

    // Calculate Coleman Law
    float coleman_law = 0.0588 * l - 0.296 * s - 15.8;
    int rounded_coleman_law = round(coleman_law);

    if (rounded_coleman_law >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (rounded_coleman_law <= -1)
    {
        printf("Before Grade 1\n");
    }
    else 
    {
        printf("Grade: %d\n", rounded_coleman_law);
    }

    return 0;
}

// Count letters
int count_letters(string text)
{
    // get length of the string
    int letters_length = strlen(text);

    for (int i = 0; i < letters_length; i++)
    {
        if (isalnum(text[i]))
        {
            letter_count++;
        }
    }

    return letter_count;
}

// Count words
int count_words(string text)
{
    // Get length of the string
    int words_length = strlen(text);

    for (int i = 0; i < words_length; i++)
    {
        if (text[i] == ' ' || text[i] == '\0')
        {
            words_count++;
        }
    }

    return words_count;
}

// count sentences
int count_sentences(string text)
{
    // get length of string
    int sentence_length = strlen(text);

    for (int i = 0; i < sentence_length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences_count++;
        }
    }
 
    return sentences_count;
}

File: problem-sets/problem-set-6/sentimental-readability/readability.py
from cs50 import get_string

# Get user input
text = get_string('Text: ')

# Count letters
letters_count = 0
def count_letters(text):
    for i in text:
        if i.isalpha():
            letters_count += 1
    return letters_count

print('Letters: ', count_letters(text))


# Count words
words_count = 1
def count_words(text):
    for i in text:
        if i == ' ':
            words_count += 1
    return words_count

print('Words: ', count_words(text))


# Count sentences
def count_sentences(text):
    sentence_count = 0
    for i in text:
        if i == '!' or i == '?' or i == '.':
            sentence_count += 1
    return sentence_count

print('Sentences: ', count_sentences(text))

# Calculate L and S
calc_l = letters_count / words_count * 100
print(calc_l)



# Call functions (C)
# Calculate L and S
# Calculate Coleman Law

# FUNCTIONS
# Count letters
    # Get length of the string
# Count words
    # Get length of the string
# Count sentences
    # Get length of the string


# Get the user's input using input() function.
# Write functions to count letters, words, and sentences. You can use similar logic to the C code you provided.
# Calculate the values of L and S using the counts obtained from step 2.
# Use the Coleman-Liau formula to calculate the index.
# Round the index to the nearest integer.
# Print the grade level according to the specifications provided.


# dinner = 'spaghetti'
# breakfast = 'cereal'
# dinner_day = 'friday'
# breakfast_day = 'Monday'

# def food(meal, day=''):
#     # return 'I ate ' + meal
#     return f'I ate {meal} on {day}'
    
# print(food(dinner, dinner_day))
# print(food(breakfast))
